### ISSUE TITLE
Some tone mapping requires the average luminance between frames

### ISSUE THREAD

**Issue by ProgramMax** [w3c/png#311]
SMPTE ST 2094-10 requires the average luminance between frames in order to apply its tone mapping.
Currently, this metadata is not provided in PNG.

(There is MaxFALL information, which is the maximum of the averages. But that is separate from the actual content average.)

If SMPTE ST 2094-10 is a tone mapping method we wish to support (which is likely), we will want a metadata chunk which carries this information.

**Comment by svgeesus** [comment]
Is there an existing standard for metadata which provides this, or does the SMPTE spec expect it to be computed on-the-fly?

Also does it use actual luminance (computed from a weighted sum of linear-light rgb values) or is it a max of gamma-encoded rgb values?

**Comment by ProgramMax** [comment]
I assume the average luminance is supposed to be pre-computed and not on-the-fly. This is only an assumption. But it seems like on-the-fly would result in the tone mapping constantly adjusting and I don't recall any smoothing in SMPTE ST 2094-10. So it could have large jumps between frames.

I have no idea if it is gamma-encoded or linear-light. If I get another look at SMPTE ST 2094-10, I'll try to check.

**Comment by digitaltvguy** [comment]
I assume the average luminance is only needed for animated PNG since MaxFALL value of one frame will be the same as the ST 2094-10 Average?  ....since the MaxFALL of a still frame would be the Max.....

**Comment by ProgramMax** [comment]
Correct.
Although, if SMPTE ST 2094-10 doesn't require MaxCLL, perhaps the cLLi chunk is a bit unwieldy even for single-frame PNGs.

**Comment by palemieux** [comment]
The average in ST 2094-10 is based on the average of code values, whereas MaxCLL is the average of linear light levels, so the two are not identical.

ST 2094-10 is not the only tone mapping algorithm, and we might wish to consider one that does not rely on the average, but only on min and max values -- which can be inferred from the ST 2086 metadata.

**Comment by ProgramMax** [comment]
It isn't within our power to choose which tone mapping method the client uses, right?

**Comment by digitaltvguy** [comment]
Should we add a reference white chunk instead of an average luminance.  This defines the level for compositing in HDR Canvas also doesn't it?

**Comment by palemieux** [comment]
> Should we add a reference white chunk instead of an average luminance?

I do not understand how this is useful and/or replaces MaxFALL ?

**Comment by digitaltvguy** [comment]
It doesn't.  It's in addition. Since China prefers a higher reference white, not documenting the reference white can cause an issue with tone-mapping (highlight compression would likely happen more aggressively above reference/graphic white).  This is how the all single-master workflows work right now in live-linear broadcasts.

**Comment by svgeesus** [comment]
> Should we add a reference white chunk instead of an average luminance? This defines the level for compositing in HDR Canvas also doesn't it?

Related to

 - https://github.com/w3c/png/issues/390

**Comment by svgeesus** [comment]
Without having a specific tone mapping algorithm in mind, it is hard to define a chunk for average luminance. Is it true luminance, or a mean of gamma-encoded code values?

@palemieux you gave examples of several methods and showed the results on a single image. 

- do you have a sense of which algorithm works well for many images?
- does that method use mean luinance?
  - if so, how is it defined?

**Comment by palemieux** [comment]
I do not think there is a single accepted method, and not all method need average luminance, e.g. _Rec. ITU-R BT.2408 (Annex 5)_ does not require such a value, and those that use average luminance do not necessarily always define it the same way.

I would avoid carrying any metadata unless there is a specified and useful use for it.

**Comment by svgeesus** [comment]
I agree with @palemieux and thus, think this issue should be closed.

If and when we recommend a specific tone mapping operation, and if that requires some metadata which is not currently representable, then we can open a new and more specific issue.

**Comment by ProgramMax** [comment]
I believe I agree.
Although we aren't trying to favor a specific tone mapping algorithm, we also shouldn't standardize every possibility.
Only after it is widely used or demanded should we standardize it. So when there is a clear requirement with enough industry backing, we'll create a new issue for that.

**Comment by ProgramMax** [comment]
To clarify a bit, if ST 2094-10 ends up widely used or demanded, we'll revisit *this specific* metadata (average luminance between frames).

If it was more general metadata that was used in many situations, that would lower the bar for us to consider it. But since this is specific, we'll wait for ST 2094-10 demand.
